### PR TITLE
feat: adapt gpg configuration to batch execution

### PR DESCRIPTION
**Pull request by jacobdotcosta**
Adapt the GPG plugin configuration so the release process, and gpg sign, can be called from a batch process

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -558,6 +558,15 @@
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-gpg-plugin</artifactId>
             <version>1.6</version>
+            <configuration>
+              <useAgent>true</useAgent>
+              <passphrase>${env.GPG_PASSPHRASE}</passphrase>
+              <gpgArguments>
+                <arg>--batch</arg>
```
Comment (cmoulliard) [inline pom.xml:565]: With maven-gpg-plugin 1.6, batch is added to the gpg command generated. So you can then remove it - see code: https://github.com/apache/maven-gpg-plugin/blob/maven-gpg-plugin-1.6/src/main/java/org/apache/maven/plugin/gpg/GpgSigner.java#L93-L100

you can test it locally
```
echo ${GPG_PASSPHRASE} | gpg --batch --passphrase-fd 0 --pinentry-mode loopback --local-user ${KEYNAME} --armor --detach-sign --no-default-keyring --output target/pom.xml.asc pom.xml
```

File: pom.xml [hunk 2]
```diff
@@ -558,6 +558,15 @@
             <groupId>org.apache.maven.plugins</groupId>
```
Comment (cmoulliard) [inline pom.xml:558]: Can you add a comment before the plugin config to explain: 
- Needed to bypass the encoding of the `passphrase` when task is executed by a job and not a human
- Syntax could change with future releases of the plugin (>= 3.0.0)

Comment (jacobdotcosta) [reply pom.xml:558]: Done.

File: pom.xml (hunk 1 above)
Comment (jacobdotcosta) [reply pom.xml]: Done.

**Comment by cmoulliard** [comment]
LGTM. Can be merged @jacobdotcosta

**Pull request merged**